File: python/OCP/step1/greeter.py
class Greeter:
    def __init__(self) -> None:
        super().__init__()
        self.__formality = "normal"

    def greet(self) -> str:
        if self.__formality == "formal":
            return "Good evening, sir."
        elif self.__formality == "casual":
            return "Sup bro?"
        elif self.__formality == "intimate":
            return "Hello Darling!"
        else:
            return "Hello."

    def set_formality(self, formality: str):
        self.__formality = formality

File: cpp/OCP/step1/greeter.cpp
#include <string>

using std::string;

class Greeter {
public:
	string Greet() {
		if (formality_ == "formal") {
			return "Good evening, sir.";
		}
		else if (formality_ == "casual") {
			return "Sup bro?";
		}
		else if (formality_ == "intimate") {
			return "Hello Darling!";
		}
		else {
			return "Hello.";
		}
	}

	void SetFormality(string formality) {
		formality_ = formality;
	}

private:
	string formality_;
};
